src/test/java/pageObjects/TagPageTests.java

```diff
@@ -0,0 +1,50 @@
+package pageObjects;
+
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Objects;
+import java.util.Random;
+
+public class TagPageTests extends BaseTest {
+
+    private String addNewTag(String newTagName) {
+        if (Objects.equals(newTagName, "")) {
+            Random rand = new Random();
+            newTagName = "tag-test-" + rand.nextInt();
+        }
+
+        tagPage.AddTag(newTagName);
+        return newTagName;
+    }
+
+    @Test
+    public void successfullyCreateTag() {
+        loginPage.signIn();
+        dashboardPage.goToTags();
+
+        String newTagName = this.addNewTag("");
```

Comment:
the steps to add a new tag should be in the TagPage class, in the TagPageTests we just want to check for the result.

Reply:
it is `tagPage.AddTag(newTagName)`.
this just follow just test case steps to create a new tag. they are part of the test